crates/jxl-render/src/filter/impls/x86_64.rs

```diff
@@ -107,3 +107,25 @@ pub fn epf_step2(
         )
     }
 }
+
+pub fn apply_gabor_like(fb: [&mut SimpleGrid<f32>; 3], weights_xyb: [[f32; 2]; 3]) {
+    if is_x86_feature_detected!("avx2") && is_x86_feature_detected!("fma") {
+        // SAFETY: Features are checked above.
+        unsafe {
+            for (fb, [weight1, weight2]) in fb.into_iter().zip(weights_xyb) {
+                run_gabor_inner_avx2(fb, weight1, weight2)
+            }
+        }
+        return;
+    }
+
+    for (fb, [weight1, weight2]) in fb.into_iter().zip(weights_xyb) {
+        super::run_gabor_inner(fb, weight1, weight2);
+    }
+}
+
+#[target_feature(enable = "avx2")]
+#[target_feature(enable = "fma")]
+unsafe fn run_gabor_inner_avx2(fb: &mut SimpleGrid<f32>, weight1: f32, weight2: f32) {
```

Comment:
Is it a placeholder or `inline` under `target_feature(enable = "avx2")` helps?

Reply:
This function is kind of isolated area for AVX2 machine code. I don't think adding `#[inline]` here helps, it's `super::run_gabor_inner` that's important to be inlined.

Reply:
Oh, I didn't realize that `is_x86_feature_detected` is a runtime check